cnd/src/network.rs

```diff
@@ -301,75 +289,41 @@ impl ComitNode {
 
     pub fn initiate_communication(
         &mut self,
-        id: LocalSwapId,
-        swap_params: Herc20HalightBitcoinCreateSwapParams,
+        local_swap_id: LocalSwapId,
+        peer: DialInformation,
+        role: Role,
+        digest: SwapDigest,
+        identities: Identities,
     ) -> anyhow::Result<()> {
-        self.supports_halight()?;
-        self.comit.initiate_communication(id, swap_params)
-    }
-
-    fn init_hbit_herc20(
-        &mut self,
-        id: LocalSwapId,
-        swap: CreatedSwap<hbit::CreatedSwap, herc20::CreatedSwap>,
-    ) -> anyhow::Result<()> {
-        self.comit.init_hbit_herc20(id, swap)
-    }
-
-    fn init_herc20_hbit(
-        &mut self,
-        id: LocalSwapId,
-        swap: CreatedSwap<herc20::CreatedSwap, hbit::CreatedSwap>,
-    ) -> anyhow::Result<()> {
-        self.comit.init_herc20_hbit(id, swap)
-    }
-
-    pub fn get_created_swap(
-        &mut self,
-        id: LocalSwapId,
-    ) -> Option<Herc20HalightBitcoinCreateSwapParams> {
-        self.comit.get_created_swap(&id)
-    }
+        match role {
+            Role::Alice => {
+                let swap_seed = self.seed.derive_swap_seed(local_swap_id);
+                let secret = swap_seed.derive_secret();
+                let secret_hash = SecretHash::new(secret);
```

Comment:
🙃  I think `secret.hash()` should be sufficient here.

Reply:
That function no longer exists because we wanted to make `Secret` independent of `SecretHash`.

Reply:
Okk, I guess I missed that 😅